src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -817,15 +817,15 @@ void DistractedSequenceRecallTestNetwork(
 
     // We increase the number of iterations (training) if the first run didn't
     // pass.
-    arma::cube inputTemp, labelsTemp;
     for (size_t iteration = 0; iteration < (9 + offset); iteration++)
     {
       for (size_t j = 0; j < trainDistractedSequenceCount; j++)
       {
+        arma::cube inputTemp, labelsTemp;
         inputTemp = arma::cube(trainInput.at(0, j).memptr(), inputSize, 1,
-            trainInput.at(0, j).n_elem / inputSize, false, true);
+            trainInput.at(0, j).n_elem / inputSize, true, true);
         labelsTemp = arma::cube(trainLabels.at(0, j).memptr(), outputSize, 1,
```

Comment:
So I ran valgrind plus gdb on this PR, one of the backtrace leads me to believe that the invalid read occurs here.

Reply:
Interesting, does the change you made to copy the auxiliary memory seem to help?  Or does it still occur despite that?

Reply:
It occurs even after creating a copy, should I run it (Valgrind + gdb) without creating a copy to see if creating a copy does help?

Reply:
I think I will do a rebase and then run valgrind + gdb again. I think the recent merge plus creating a copy might help, not sure though.

Reply:
:+1: so, one of the things that could be happening here that could make valgrind point out this line in specific is if some memory gets allocated to make `inputTemp` here, but then that memory is, e.g., passed somewhere else but `inputTemp` goes out of scope and the memory gets freed.

I'm not sure if that's happening here though.  I hope that makes sense, here's a MWE:

```
int main()
{
  arma::mat alias;

  {
    arma::mat data(10, 10, arma::fill::randu);
    alias = arma::mat(data.memptr(), 10, 10, false, true);
  }

  // This is invalid!  `data` has been freed and so `alias.memptr()` is no longer valid.
  std::cout << alias;
}
```

Anyway, my best guess is something like that is happening somewhere in the code.  Tracking down exactly where or how is pretty hard given the amounts of indirection in the RNN code though. :smile:

Reply:
Agreed that does make sense, I will try to get track where variables are assigned logically and also using valgrind. Hopefully I find something that fixes this.

Reply:
Hey @rcurtin, sorry for the delayed response, but I am at loss with why the invalid read occurs. I have tried creating a copy anywhere that might lead to an error. I went through how the function calls were made and incase we were using variables that were not accessible, I didn't find any but I am going through them again 😔. Any suggestions on how do I go about investigating this further. I have been running valgrind for the past couple of days with no results to show 😔. I would really appreciate any help or suggestions. Thanks.